### DISCUSSION TITLE
fail to install on PHP8

### DISCUSSION THREAD

**Discussion by elfeffe** [spatie/laravel-ray#180] [Q&A]
I just try to require as usual, but instead of php7.4, I'm trying with php8.

It fails because it requires macroable1.0, and macroable1.0 requires php7.

**Comment by freekmurze** [comment]
Our tests indicate that it is perfectly installable under PHP 7.4
https://github.com/spatie/laravel-ray/actions/runs/723150426

Could you try to install it in a fresh project?

**Reply by elfeffe** [reply]
But I'm saying this is not installable under PHP8.
It says it needs macroable 1, but macroable 2 is compatible with PHP8

**Reply by patinthehat** [reply]
@elfeffe Can you post the error you're seeing (and your OS, PHP version, etc.)? 
I am able to install `spatie/laravel-ray` with PHP 8 without a problem.

**Reply by elfeffe** [reply]
composer require spatie/laravel-ray
Using version ^1.17 for spatie/laravel-ray
./composer.json has been updated
Running composer update spatie/laravel-ray
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - spatie/laravel-ray[1.17.0, ..., 1.17.2] require spatie/ray ^1.21.2 -> satisfiable by spatie/ray[1.21.2].
    - spatie/ray 1.21.2 requires spatie/macroable ^1.0 -> found spatie/macroable[1.0.0, 1.0.1] but the package is fixed to 2.0.0 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - Root composer.json requires spatie/laravel-ray ^1.17 -> satisfiable by spatie/laravel-ray[1.17.0, 1.17.1, 1.17.2].

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.

**Reply by patinthehat** [reply]
Can you try removing the lock file and vendor directory and running `composer install`?
You can also try `composer require spatie/laravel-ray -W`.

**Reply by elfeffe** [reply]
The full trace:

```
rm composer.lock
root@ubuntu-server:/web/# composer require spatie/laravel-ray -W
Using version ^1.17 for spatie/laravel-ray
./composer.json has been updated
Running composer update spatie/laravel-ray --with-all-dependencies
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - spatie/macroable 1.0.0 requires php ^7.0 -> your php version (8.0.3) does not satisfy that requirement.
    - spatie/laravel-ray[1.17.0, ..., 1.17.2] require spatie/ray ^1.21.2 -> satisfiable by spatie/ray[1.21.2].
    - spatie/ray 1.21.2 requires spatie/macroable ^1.0 -> satisfiable by spatie/macroable[1.0.0, 1.0.1].
    - You can only install one version of a package, so only one of these can be installed: spatie/macroable[1.0.0, 1.0.1, 2.0.0].
    - spatie/url 2.0.0 requires spatie/macroable ^2.0 -> satisfiable by spatie/macroable[2.0.0].
    - Root composer.json requires spatie/url ^2.0 -> satisfiable by spatie/url[2.0.0].
    - Root composer.json requires spatie/laravel-ray ^1.17 -> satisfiable by spatie/laravel-ray[1.17.0, 1.17.1, 1.17.2].


Installation failed, reverting ./composer.json to its original content.

```

Can I share anything else?

**Comment by sarmenb** [comment]
im also running php 8.0.3 and ray is running perfectly. What i recommend you try is updating ray as well as laravel and see if that fixes the issue.

**Reply by elfeffe** [reply]
Is anybody using spatie/url together with Ray?
That seems to be the problem.

**Reply by elfeffe** [reply]
@freekmurze The problem is that it requires spatie/ray (of course) and spatie/ray requires macroable 1, wich is not compatible with PHP8
`    - spatie/macroable 1.0.0 requires php ^7.0 -> your php version (8.0.3) does not satisfy that requirement.`

https://github.com/spatie/ray/blob/master/composer.json

And I have spatie/url , which requires macroable 2, and it works with PHP8
Anyway without spatie/url, it seems that macroable 1 requires php7, so it will not install in PHP8 (I know some of you can, but it can be that you have php7 for your cli and 8 for fpm).
Do you plan to require macroable 2 for spatie/ray soon?

**Reply by freekmurze** [reply]
My colleague @Nielsvanpach will look into this 👍

**Reply by Nielsvanpach** [reply]
In the latest release of spatie/ray you can now use macroable v2

**Reply by elfeffe** [reply]
I confirm it works now, thank you
